packages/donor/src/screens/extendedSignup/PersonalDetails.tsx

```diff
@@ -129,7 +155,7 @@ export default function PersonalDetails(props: PersonalDetailsProps) {
         <Button
           onClick={onSave}
           title={props.buttonText}
-          isDisabled={!areAllFieldsValid}
+          isDisabled={!areAllFieldsValid || !allFieldsHaveValue}
```

Comment:
This PR looks great, and I looked at the Storybook of it, and saw the error shows when I type the first letter, and disappears when I have more than 2, same for the phone number.

I have a suggestion on how to make the experience better:
1. Always enable the save button.
2. On save - check all values
3. If valid - proceed.
4. If not - show errors when needed and don't proceed.

What do you think?

Reply:
Emmm personally I don't like forms with that behavior because you to go backwards and fill it again and it feels like you are filling it twice, on the other hand it is a very short form. Are you aware of any plans to expand it ? @yaron1m

Reply:
We do plan to add a field or 2 in the future.

Anyway - I trust your judgment. If you think it is a better experience this way - I'm in